### PR TITLE
Pwa 2291: Streamline Cypress tests and create documentation

### PR DESCRIPTION
**Pull request by DiegoBezerra89**
## Description

Streamline the Cypress tests, changed the selectors based on classes to `[data-cy="element"]` pattern.
Updated some file names, as folder structures.
Creation of a documentation to explain how Cypress was applied on project.

## Related Issue

PWA - 2291: Streamline Cypress tests and create documentation

Closes PWA-2291.

#### Test scenario(s) for direct fix/feature

Run all Cypress Tests on `venia-integration-tests`

## Checklist

-   ~~I have added tests to cover my changes, if necessary.~~
-   ~~I have added translations for new strings, if necessary.~~
-   I have updated the documentation accordingly, if necessary.

### CODE REVIEW

File: venia-integration-tests/src/fields/miniCart/index.js [hunk 1]
```diff
@@ -13,4 +19,19 @@ export const guestElementSelector = '[data-cy="GuestForm-root"]';
 export const miniCartProductLink =
     '[data-cy="MiniCart-body"] [data-cy="item-name"]';
 
+<<<<<<< HEAD
```
Comment (fooman) [inline venia-integration-tests/src/fields/miniCart/index.js]: Still some merge information here

**Review by dpatil-magento — COMMENTED** [review]
Pls take care of prettier and lint formatting test failures.

File: venia-integration-tests/cypress.config.json [hunk 2]
```diff
@@ -1,5 +1,5 @@
 {
-    "baseUrl": "https://develop.pwa-venia.com/",
+    "baseUrl": "https://magento-venia-concept-7nbzp.local.pwadev:8708/",
```
Comment (dpatil-magento) [inline venia-integration-tests/cypress.config.json:2]: @DiegoBezerra89 Please revert it to develop.

File: venia-integration-tests/README.md [hunk 3]
```diff
@@ -47,5 +47,55 @@ In this category we add Cypress tests for any End to End user flows based on Mag
 ## Venia Integration Tests
 In this category we add Cypress tests for user flows which are on Magento non-default configurations. These tests run on mocked data. Please refer to [multiple wishlist][] test as an example. This example test is a very large test given auth user session restriction, but we prefer to have these mock data tests as small as possible.
 
-[single wishlist]: https://github.com/magento/pwa-studio/blob/develop/venia-integration-tests/src/tests/wishList/e2eTests/singleWishlistAddRemoveProduct.spec.js
-[multiple wishlist]: https://github.com/magento/pwa-studio/blob/develop/venia-integration-tests/src/tests/wishList/integrationTests/verifyMultipleWishlistFeatures.spec.js
+[single wishlist]: https://github.com/magento/pwa-studio/blob/develop/venia-integration-tests/src/tests/e2eTests/wishList/singleWishlistAddRemoveProduct.spec.js
+[multiple wishlist]: https://github.com/magento/pwa-studio/blob/develop/venia-integration-tests/src/tests/integrationTests/wishList/verifyMultipleWishlistFeatures.spec.js
+
```
Comment (dpatil-magento) [inline venia-integration-tests/README.md:52]: @DiegoBezerra89 Can you please add note about `packages/venia-concept/.env` need to have variable set `BABEL_KEEP_ATTRIBUTES=true` in order load `data-cy` attributes on test deployed instances?

File: venia-integration-tests/cypress.config.json (hunk 2 above)
Comment (DiegoBezerra89) [reply venia-integration-tests/cypress.config.json:2]: Sorry...

**Comment by dpatil-magento** [comment]
@DiegoBezerra89  just move these two test and its good to go for QA
`venia-integration-tests/src/tests/e2eTests/categories/guestCategoryAccess.spec.js -> venia-integration-tests/src/tests/e2eTests/category/guestCategoryAccess.spec.js`
`venia-integration-tests/src/tests/e2eTests/cartPage/editCartAndCheckout.spec.js -> venia-integration-tests/src/tests/e2eTests/checkout/editCartAndCheckout.spec.js`
And delete `e2eTests/categories` and `e2eTests/cartPage` directories (after moving tests)

**Comment by dpatil-magento** [comment]
QA Approved

**Review by dpatil-magento — DISMISSED** [review]

**Comment by dpatil-magento** [comment]
Re-ran sign-in, account info and registered user tests. Looks good

**Review by dpatil-magento — APPROVED** [review]

**Pull request merged**